scipy/stats/_sobol.pyx

```diff
@@ -240,16 +271,16 @@ cpdef void _initialize_v(
 
 @cython.boundscheck(False)
 @cython.wraparound(False)
-cpdef void _draw(const cnp.uint64_t n,
-                 const cnp.uint64_t num_gen,
+cpdef void _draw(const uint_32_64 n,
```

Comment:
I'm guessing a bit here, but I wonder if `n`, `num_gen`, `scale` are causing the problem. Where they're called from Python they're called with a Python integer for `n`. I'm not sure which int type that'll select, but I think that'll set the type required for `sv` and the array arguments.

Maybe try leaving the arguments untyped (as Python types) as then do `cdef uint_32_64 n = n_py` (where `n_py` is the argument) inside the function.

That will hopefully let Cython select the type based on the type of the array arguments.

Reply:
That's exactly the issue! I tried to remove the types from all three (and the `nogil` then) and all tests are passing now 😃 

Shall I do two functions then? A `def` wrapping a `cdef`?

Reply:
A `def` wrapping a `cdef` would be the way to do it. It's only worthwhile creating the extra wrapper layer if you need to call it efficiently from within Cython (otherwise just make the `def`)

Reply:
I used a wrapper since it was 10x slower otherwise.